### DISCUSSION TITLE
CPU spikes reported by task manager, normal behavior?

### DISCUSSION THREAD

**Discussion by manatactical** [apohl79/audiogridder#949] [General]
Hi,

This is clearly not a bug, and AG has been working well despite this, but: I've noticed for quite some time now that my windows task manager reports some spike when AG is running (both idle and playing). _(I've allocated AG on 4 cores, 12 to 15.)_
I have not yet maxed out those cores to see if those tiny spikes affect the audio, but well, it's worth reporting. (other cores are running Vienna Ensemble Pro et Cantabile).

![IMG_20220517_093124](https://user-images.githubusercontent.com/9537424/168964451-67b60d8e-993d-4e59-a924-ebe9aa2fc044.jpg)
)

**Comment by apohl79** [comment]
Thx for reporting this. Unfortunately I can't reproduce this, so I need more info. 
- Can you describe your setup a bit more? 
- Which plugins have you loaded on how many AG instances? If you bypass the plugins in AG, do the spikes go away? Is it a specific plugin maybe causing this?
- Are you using local mode or is it a remote server? 
- Are you streaming a plugin UI if not in local mode?
- Are you using sandboxing (chain or plugin mode?) 
- Maybe you can also try to bypass AG plugins in the DAW and see if the issue is caused by a particular instance.

**Comment by manatactical** [comment]
Sure! 

I'm running two machines. macOS for the plugin side and PC for the server-side (10980XE, 192GB of RAM, SSDs)
- I was using a few instances of AG, 6-7 or so.
- I was running SINE Player and Kontakt. So does VEP.
- I was using it as a remote server, and I don't stream anything.
- The server is set on sandbox mode yes (chain).

I'm going to try the bypass thing tomorrow morning. I will keep you posted.

**Comment by manatactical** [comment]
Okay,

The CPU spikes were due to a peak of ethernet transfer caused by one of the Kontakt libraries. Not sure why. So good news!
However, I've also found another "bug". When you load an instrument from Kontakt, as the audio transfer is inactive because no MIDI has been received yet, the GUI is glitched (a common issue with Kontakt when the audio engine is off).
And it goes like that every time MIDI activity fades out.

Maybe you can keep the "audio transfer" active when the plugin view is opened?

**Reply by apohl79** [reply]
Ok thx for sorting it out!

> Maybe you can keep the "audio transfer" active when the plugin view is opened?

Yes no prob.
